src/loctool/PendoXliffFileType.ts

```diff
@@ -0,0 +1,347 @@
+        // except that localeMap is in root "settings" section rather than in the plugin-specific one
+        // @ts-expect-error -- undocumented
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- undocumented
+        const localeMap = this.project.settings?.localeMap as unknown;
+
+        // if there is a locale mapping, use it
+        if (
+            localeMap &&
+            typeof localeMap === "object" &&
+            loctoolLocale in localeMap &&
+            // @ts-expect-error -- undocumented
+            typeof localeMap[loctoolLocale] === "string" &&
+            // @ts-expect-error -- undocumented
+            localeMap[loctoolLocale].length > 0
+        ) {
+            // @ts-expect-error -- undocumented
+            return localeMap[loctoolLocale];
+        }
+
+        return loctoolLocale;
+    }
+
+    /**
+     * Wraps translation units in loctool's ResourceString objects.
+     */
+    private createTranslationSet(units: TranslationUnit[]) {
+        // convert to loctool resources
+        const resources = units.map((unit) =>
+            this.loctoolAPI.newResource({
+                resType: "string",
+                key: unit.key,
+                sourceLocale: unit.sourceLocale,
+                source: unit.source,
+                comment: unit.comment,
+                // not sure if this is used by loctool anywhere,
+                // but it's required per the Resource interface definition
+                project: this.project.getProjectId(),
+            }),
+        );
+
+        // wrap in a translation set
+        const translationSet = this.loctoolAPI.newTranslationSet<ResourceString>(this.project.getSourceLocale());
+        translationSet.addAll(resources);
+
+        return translationSet;
+    }
+
+    getExtracted(): TranslationSet {
+        // this should ouput a translationset with merged resources from all files
+        // which fit this filetype;
+        // note: this means that the instance of the filetype class has to keep track of all files it
+        // has processed (i.e. each {@link newFile} call should add the file to a list)
+
+        const translationSet = this.loctoolAPI.newTranslationSet(this.sourceLocale);
+
+        for (const path in this.files) {
+            const file = this.files[path];
+            translationSet.addSet(file.getTranslationSet());
+        }
+
+        return translationSet;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars -- noop, see comment in method
+    addSet(set: TranslationSet): void {
+        // no-op
+        // it's not clear why would the filetype need to add a set of translations to itself
+        // given that per the getExtracted method, sets should come from the files
+    }
+
+    getNew(): TranslationSet {
+        // not sure how this differs from getExtracted
```

Comment:
It is only the new strings. ie. the ones that did not have translations during `localize()`. This is more important when using loctool without mojito which figures out what is new by itself.

Reply:
Similar to pseudotranslations, I think segregation of new vs old strings should be handled by loctool.